lib/rpc/globalRpcProviders.ts

```diff
@@ -0,0 +1,62 @@
+import { ChainId } from '@uniswap/sdk-core'
+import SingleJsonRpcProvider from './singleJsonRpcProvider'
+import UniJsonRpcProvider from './uniJsonRpcProvider'
+
+export default class GlobalRpcProviders {
+  private static readonly PROVIDER_RPC_URL_RANKING: Partial<Record<ChainId, number[] | undefined>> = {
```

Comment:
It feels like we can just use `Map` construct instead of `Partial<Record>`. There are places in routing-api where we use Map, like https://github.com/Uniswap/routing-api/blob/main/lib/handlers/quote/util/pairs-to-track.ts